src/ui/public/agg_types/buckets/histogram.js

```diff
@@ -27,8 +35,52 @@ export function AggTypesBucketsHistogramProvider(Private) {
       {
         name: 'interval',
         editor: intervalTemplate,
+        onSearchRequestStart(aggConfig, searchSource) {
+          const field = aggConfig.getField();
+          const aggBody = field.scripted
+            ? { script: { inline: field.script, lang: field.lang } }
+            : { field: field.name };
+
+          return searchSource
+            .extend()
+            .size(0)
+            .aggs({
+              maxAgg: {
+                max: aggBody
+              },
+              minAgg: {
+                min: aggBody
+              }
+            })
+            .fetchAsRejectablePromise()
+            .then((resp) => {
+              aggConfig.autoBounds = {
```

Comment:
Should the getters/setter methods really be added to AggConfig since most other configs will never need `autoBounds`? Autobounds is added to AggConfig as a decoration https://github.com/nreese/kibana/blob/hist_pre_flight/src/ui/public/agg_types/buckets/histogram.js#L22.

Reply:
I don't think this should go on the `aggConfig` at all, but should go on `aggConfig.params`. That's what the params object is for, and adding a getter/setter to a general class like `AggConfig` doesn't really make sense to me.

Reply:
Oh, I guess you could just update your `decorateAggConfig` function with `getAutoBounds()` and `setAutoBounds()`...